pysmt/logics.py

```diff
@@ -594,11 +594,19 @@ def __hash__(self):
 
 def get_logic_by_name(name):
     """Returns the Logic that matches the provided name."""
-
     for logic in LOGICS:
-        if logic.name == name: return logic
+        if logic.name.lower() == name.lower(): return logic
     raise UndefinedLogicError(name)
 
+def convert_logic_from_string(name):
+    """Helper function to parse function arguments.
+
+    This takes a logic or a string or None, and returns a logic or None.
+    """
+    if name is not None and type(name) == str:
```

Comment:
In PY2 we need to include unicode as a possible type. This is not a problem for PY3.